src/core/viewports/viewport.js

```diff
@@ -303,7 +303,15 @@ export default class Viewport {
     return false;
   }
 
-  getDistanceScales() {
+  getDistanceScales(coordinateOrigin) {
+    if (coordinateOrigin) {
+      return getDistanceScales({
+        longitude: coordinateOrigin[0],
+        latitude: coordinateOrigin[1],
+        scale: this.scale,
+        highPrecision: true
+      });
+    }
     return this.distanceScales;
```

Comment:
Thought: How valuable is it to store a precalculated distance scales, will it be used often or better to drop pre-calculated scales and only calculate on demand?

Reply:
The pre-calculated `distanceScales` are still used for distance and normal projection for all layers. 
`coordinateOrigin` only applies to offset modes.